File: typescript/darts/darts.ts
export function score(x: number, y: number): number {
  let distanceFromCenter = Math.sqrt((x * x) + (y * y));

  switch(true) {
    case distanceFromCenter <= 1:
      return 10;
      break;
    case distanceFromCenter > 1 && distanceFromCenter <= 5:
      return 5;
      break;
    case distanceFromCenter > 5 && distanceFromCenter <= 10:
      return 1;
      break;
    default: 
      return 0;
  }
}

File: cpp/darts/darts.cpp
#include <cmath> 
#include "darts.h"

namespace darts {
  int score(double x, double y) {
    double distance = sqrt(x*x + y*y);

    if(distance <= 1) { return 10; }
    else if (distance > 1 && distance <= 5) { return 5; }
    else if (distance > 5 && distance <= 10) { return 1; }
    else { return 0; }
  }
} // namespace darts